### ISSUE TITLE
Image.select() doesn't work correctly

### ISSUE THREAD

**Issue by Prgckwb** [gradio-app/gradio#9297]
### Describe the bug

I am trying to obtain coordinate information of an image using the Image.select() listener, but the select function does not work in the latest version. 

Not work on  'gradio==4.43.0'
But work on 'gradio==3.50.2'

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

```python
import gradio as gr

with gr.Blocks() as demo:
    with gr.Row():
        image = gr.Image(label="Image", interactive=True)
        text = gr.Textbox(label="Text", interactive=True)

    def on_select(evt: gr.SelectData):
        print(evt)
        return f"You selected {evt.value} at {evt.index} from {evt.target}"

    image.select(on_select, outputs=text)
    image.clear(on_select, outputs=text)

demo.launch()
```

### System Info

```shell
Gradio Environment Information:
------------------------------
Operating System: Darwin
gradio version: 4.43.0
gradio_client version: 1.3.0

------------------------------------------------
gradio dependencies in your environment:

aiofiles: 23.2.1
anyio: 4.4.0
fastapi: 0.112.4
ffmpy: 0.4.0
gradio-client==1.3.0 is not installed.
httpx: 0.27.2
huggingface-hub: 0.24.6
importlib-resources: 6.4.4
jinja2: 3.1.4
markupsafe: 2.1.5
matplotlib: 3.9.2
numpy: 1.26.4
orjson: 3.10.7
packaging: 24.1
pandas: 2.2.2
pillow: 10.4.0
pydantic: 2.9.0
pydub: 0.25.1
python-multipart: 0.0.9
pyyaml: 6.0.2
ruff: 0.6.4
semantic-version: 2.10.0
tomlkit==0.12.0 is not installed.
typer: 0.12.5
typing-extensions: 4.12.2
urllib3: 2.2.2
uvicorn: 0.30.6
authlib; extra == 'oauth' is not installed.
itsdangerous; extra == 'oauth' is not installed.


gradio_client dependencies in your environment:

fsspec: 2024.6.1
httpx: 0.27.2
huggingface-hub: 0.24.6
packaging: 24.1
typing-extensions: 4.12.2
websockets: 11.0.3
```

### Severity

I can work around it

**Comment by abidlabs** [comment]
Can confirm the issue, thanks @Prgckwb

**Comment by abidlabs** [comment]
cc @aliabid94 if you can take a look?

**Comment by abidlabs** [comment]
Actually duplicate of https://github.com/gradio-app/gradio/issues/9182, so let me close in favor of that issue

**Comment by iw17** [comment]
> Actually duplicate of #9182, so let me close in favor of that issue

What? You closed #9182 in favor of this issue, and closed this one saying it was a dup of #9182? Is that why it's not had any update or fix?

**Comment by abidlabs** [comment]
9182 is still open
